javascript-forms/form-events.js

```diff
@@ -0,0 +1,50 @@
+const $userName = document.getElementById('user-name');
+console.log($userName);
+
+const $userEmail = document.getElementById('user-email');
+console.log($userEmail);
+
+const $userMessage = document.getElementById('user-message');
+console.log($userMessage);
+
+$userName.addEventListener('focus', function handleFocus(event) {
+  console.log('focus event fired');
+  console.log('event.target.name:', event.target.name);
+});
+
+$userName.addEventListener('blur', function handleFocus(event) {
+  console.log('blur event fired');
+  console.log('event.target.name:', event.target.name);
+});
+
+$userName.addEventListener('input', function handleFocus(event) {
+  console.log('event.target.name:', event.target.name, event.target.value);
+});
+
+$userEmail.addEventListener('focus', function handleBlur(event) {
+  console.log('blur event fired');
+  console.log('event.target.name:', event.target.name);
+});
+
+$userEmail.addEventListener('blur', function handleBlur(event) {
+  console.log('blur event fired');
+  console.log('event.target.name:', event.target.name);
+});
+
+$userEmail.addEventListener('input', function handleBlur(event) {
+  console.log('event.target.name:', event.target.name, event.target.value);
+});
+
+$userMessage.addEventListener('focus', function handleInput(event) {
+  console.log('focus event fired');
+  console.log('event.target.name:', event.target.name);
+});
+
+$userMessage.addEventListener('blur', function handleInput(event) {
+  console.log('blur event fired');
+  console.log('event.target.name:', event.target.name);
+});
+
+$userMessage.addEventListener('input', function handleInput(event) {
+  console.log('event.target.name:', event.target.name, event.target.value);
+});
```

Comment:
Instead of creating all these functions inside of your `addEventListener` you should be creating the functions on their own. So you should have 3 functions `handleFocus`, `handleInput`, & `handleBlur`. Then you'll pass those functions into your `addEventListener` methods as a callback function.

So only 3 functions should be defined, and then each form element should have 3 different event listeners attached to it